client-course-schedulizer/src/utilities/helpers/caseFunctions.ts

```diff
@@ -27,12 +27,36 @@ const thursReg = RegExp("[Tt][Hh]|[Rr]");
 const friReg = RegExp("[Ff]");
 const satReg = RegExp("[Ss](?![Uu])");
 
-export const startTimeCallback = (value: string, { firstMeeting }: CaseCallbackParams) => {
-  firstMeeting.startTime = startTimeCase(value);
+const createMeetings = (value: string, { meetings }: CaseCallbackParams): string[] => {
+  const valueParts = value.split("\n");
+  for (let i = 0; i < valueParts.length; i += 1) {
+    if (meetings.length <= i) {
+      // If there aren't enough meetings, create a new one
+      meetings.push({
+        days: [],
+        duration: 0,
+        location: { building: "", roomCapacity: 0, roomNumber: "" },
+        startTime: "",
+      });
+    }
+  }
+  return valueParts;
+};
+
+export const startTimeCallback = (value: string, params: CaseCallbackParams) => {
+  const startTimes = createMeetings(value, params);
+  const { meetings } = params;
+  startTimes.forEach((startTime, i) => {
+    meetings[i].startTime = startTimeCase(startTime);
+  });
```

Comment:
I see this pattern of logic repeated often. Would it be better to make a function helper?

```
/* 
A function to createMeeting and assign them using a custom assignment function.
*/
const assignWithMeetings = (
  value: string,
  params: CaseCallbackParams,
  iterator: (value: string, i: number, arr: Meeting[]) => void,
) => {
  const valueParts = createMeetings(value, params);
  const { meetings } = params;
  valueParts.forEach((v, i) => {
    iterator(v, i, meetings);
  });
};
```

Use: 
```
export const startTimeCallback = (value: string, params: CaseCallbackParams) => {
  assignWithMeetings(value, params, (startTime, i, meetings: Meeting[]) => {
    meetings[i].startTime = startTimeCase(startTime);
  });
};

export const locationCallback = (value: string, params: CaseCallbackParams) => {
  assignWithMeetings(value, params, (location, i, meetings: Meeting[]) => {
    [meetings[i].location.building, meetings[i].location.roomNumber] = locationCase(location);
  });
};
```
Maybe this is better because it allows for this pattern to be reused easily. Let me know what you think.

Reply:
This is awesome!

Reply:
This is great, thanks: 2453811